pokemongo_bot/cell_workers/bot_threshold.py

```diff
@@ -0,0 +1,125 @@
+from datetime import datetime, timedelta
+from time import sleep
+from random import uniform, randint
+from pokemongo_bot.base_task import BaseTask
+
+class BotThreshold(BaseTask):
+    """Pauses the execution of the bot after some configured threshold for anti-detection purposes
+
+    Example Config:
+    {
+        "type": "BotThreshold",
+        "config": {
+          "enabled": true,
+          "duration":"18:30",
+          "duration_random_offset": "00:30"
+          "min_pokemon_caught": 500,
+          "max_pokemon_caught": 900,
+          "min_pokestop_spun": 1000,
+          "max_pokestop_spun": 1900
+        }
+    }
+    duration: (HH:MM) the duration of rest time
+    duration_random_offset: (HH:MM) random offset of duration of rest
+                        for this example the possible duration is 18 hours to 19 hours
+    min_pokemon_caught: (0-inf) minimum number of pokemon to capture before going to rest
+    max_pokemon_caught: (0-inf) maximum number of pokemon to capture before going to rest
+    min_pokestop_spun: (0-inf) minimum number of pokestops to spin before going to rest
+    max_pokestop_spun: (0-inf) minimum number of pokestops to spin before going to rest
+    """
+    SUPPORTED_TASK_API_VERSION = 1
+    LOG_INTERVAL_SECONDS = 600
+
+    def initialize(self):
+        self._process_config()
+        self._calculate_threshold()
+
+    def work(self):
+        if self._should_rest_now():
+            self._rest()
+            raise Exception("Resetting metrics for next run of BotThreshold")
```

Comment:
@cwild That's the line I was referring to in my PR above. Thanks for pointing it out :)